File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    int answer = x;
    int sum = 0;

    while(x != 0){
        sum += x % 10;
        x /= 10;
    }
    
    if(answer % sum == 0) return true;
    else return false;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    let answer = x;
    let sum = 0;

    while(x != 0){
        sum += Math.floor(x % 10);
        x = Math.floor(x / 10);
    }
    
    if(answer % sum == 0) return true;
    else return false;
}